Common/Logger.swift

```diff
@@ -0,0 +1,62 @@
+//
+//  Logger.swift
+//  Ion
+//
+//  Created by Anton K on 09/10/2019.
+//
+
+final public class Logger<T>: EmitterDelegate {
+
+    public var name: String?
+    public var isEnabled: Bool = true
+    weak var target: EmitterDelegate?
+
+    public init() {}
+
+    private func pointer(to object: AnyObject) -> String {
+        return "\(Unmanaged.passUnretained(object).toOpaque())"
+    }
+
+    public func emitter<T>(_ sender: Emitter<T>, didRegister subscriber: Subscriber<T>) {
+        if isEnabled {
+            print("[II] E[\(name ?? pointer(to: sender))] didRegister: \(pointer(to: subscriber))")
+        }
+        target?.emitter(sender, didRegister: subscriber)
```

Comment:
Could you unify logging logic to reduce code duplication? I mean log format and `name ?? pointer(to: sender)`